src/main/java/jenkins/branch/RateLimitBranchProperty.java

```diff
@@ -533,23 +533,22 @@ public CauseOfBlockage canRun(Queue.Item item) {
                     }
                     // ensure items leave the queue in the order they were scheduled
                     List<Queue.Item> items = Jenkins.get().getQueue().getItems(item.task);
-                    if (items.size() == 1 && item == items.get(0)) {
-                        return null;
-                    }
-                    for (Queue.Item i : items) {
-                        if (i.getId() < item.getId()) {
-                            LOGGER.log(Level.FINE, "{0} with queue id {1} blocked by queue id {2} was first",
-                                    new Object[]{
-                                            job.getFullName(),
-                                            item.getId(),
-                                            i.getId()
-                                    }
-                            );
+                    if (!items.isEmpty()) {
+                        Queue.Item i = items.get(0);
+                        if (item == i) { // This item is the first in queue for this task.
+                            LOGGER.log(Level.FINE, () -> "Allowing " + item.getId() + " " + item.getClass().getSimpleName() + " " + job.getFullName());
+                            return null;
+                        } else {
+                            // There is another item before the one currently processed, so block.
+                            LOGGER.log(Level.FINE, () -> item.getId() + " " + job.getFullName() + " blocked by queue id " + i.getId() + " was first");
```

Comment:
This branch has no test coverage. It is not exactly clear to me the circumstances under which it would be activated—a parameterized build with multiple (distinct) items in queue? The former logic does not look reliable (there is no documentation saying `Queue.Item.getId` is comparable) but neither does this (there is no documentation indicating an order of `Queue.getItems(Queue.Task)`).

The logic being edited was added in 4ed032736012892bcee8b93c7128af84d1c66dd8 (#58) and added `RateLimitBranchPropertyTest.rateLimitsConcurrentBuilds`. With logging from this PR but logic from `master`, that test does print

```
FINE	j.b.RateLimitBranchProperty$QueueTaskDispatcherImpl#canRun: foo/master with queue id 4 blocked by queue id 3 was first
```

At that point `items` contains 4 then 3 in that order and we are checking 4, so the desired behavior is to block it.

I think what this should really be doing is closer to the original logic, but using `getInQueueSince()` rather than `getId()`.